textproto/header.go

```diff
@@ -62,6 +62,30 @@ func newHeader(fs []*headerField) Header {
 	return Header{l: fs, m: m}
 }
 
+// AddRaw adds the raw key, value pair to the header. The supplied string
+// should be a complete field in the Key: Value form including trailing CRLF.
+// No changes are made to it and it will be inserted into WriteHeader output as
+// is.
+//
+// kv is directly added to the underlying structure and therefore should not be
+// modified after the AddRaw call.
+func (h *Header) AddRaw(kv []byte) {
+	colon := bytes.IndexByte(kv, ':')
+	if colon == -1 {
+		panic("textproto: Header.AddRaw: missing colon")
```

Comment:
This should be documented (or this function should return an error? Not sure what's best)

Reply:
>or this function should return an error? Not sure what's best.

`panic` is usually used to indicate bugs in the user code as opposed to `error` used for handling of expected errors.

With AddRaw, there are a lot of ways to provide meaningless input and have it directly copied by WriteHeader to the io.Writer, resulting in corrupted headers. Therefore, there is an implicit assumption that the user code takes responsibility for making sure the input is well-formed. If it does not - it is a bug.

Additionally, if we go with `error`, it is prone to be ignored in most cases since it fails only in the certain specific case "that obviously will not happen". This will make bugs harder to spot.

Reply:
Works for me